tests/set_maintenance/test_set_maintenance.py

```diff
@@ -0,0 +1,178 @@
+import pytest
+from mock import patch
+
+from pubtools.pulplib import (
+    FakeController,
+    Distributor,
+    Repository,
+    FileRepository,
+    Task,
+    PulpException,
+)
+
+from pubtools._pulp.tasks.set_maintenance import SetMaintenanceOn, SetMaintenanceOff
+
+
+@pytest.fixture
+def mock_on_logger():
+    with patch(
+        "pubtools._pulp.tasks.set_maintenance.set_maintenance_on.LOG"
+    ) as mocked_info:
+        yield mocked_info
+
+
+@pytest.fixture
+def mock_off_logger():
+    with patch(
+        "pubtools._pulp.tasks.set_maintenance.set_maintenance_off.LOG"
+    ) as mocked_info:
+        yield mocked_info
+
+
+def maintenance_repo():
+    iso_distributor = Distributor(id="iso_distributor", type_id="iso_distributor")
+    repo = FileRepository(id="redhat-maintenance", distributors=[iso_distributor])
+
+    return repo
+
+
+def _get_fake_controller(*args):
+    controller = FakeController()
+    for repo in args:
+        controller.insert_repository(repo)
+    controller.insert_repository(maintenance_repo())
+    return controller
+
+
+def _patch_pulp_client(client):
+    return patch("pubtools._pulp.services.PulpClientService.pulp_client", client)
+
+
+def _run_test(cmd_args, on, *repos):
+    controller = _get_fake_controller(*repos)
+    if on:
+        s_m = SetMaintenanceOn()
+    else:
+        s_m = SetMaintenanceOff()
+    arg = ["", "--pulp-url", "http://some.url", "--verbose"]
+    arg.extend(cmd_args)
+
+    with patch("sys.argv", arg):
+        with _patch_pulp_client(controller.client):
+            s_m.main()
+    return controller
+
+
+def test_maintenance_on(mock_on_logger):
+    """Test set maintenance by passing repo ids."""
+    repo1 = Repository(id="repo1")
+    repo2 = Repository(id="repo2")
+
+    cmd_args = ["--repo-id", "repo1", "repo2", "--message", "Now in Miantenance"]
+
+    controller = _run_test(cmd_args, True, repo1, repo2)
+
+    # upload and publish should be called once each
+    assert len(controller.upload_history) == 1
+    assert controller.upload_history[0].repository.id == "redhat-maintenance"
+    assert len(controller.publish_history) == 1
+    assert controller.publish_history[0].repository.id == "redhat-maintenance"
+
+    # logged message should indicate both repos are set to maintenance
```

Comment:
1. Instead of testing logged messages manually, can you please consider using the command_tester fixture which would test the entire log at once?  This would protect against e.g. completely bogus log messages left in the code accidentally, and also let code reviewers see how the entire task is expected to look in practice.

2. If you don't want to do that, instead of mocking the logger, can you please consider using caplog fixture instead?  This would at least let the test keep working without depending on that specific logger and method call being used.